app/screens/scheduled_post_options/scheduled_post_picker.tsx

```diff
@@ -0,0 +1,148 @@
+    const [customTimeSelected, setCustomTimeSelected] = useState(false);
+    const userTimezone = getTimezone(currentUserTimezone);
+
+    const style = getStyleSheet(theme);
+
+    const snapPoints = useMemo(() => {
+        const bottomSheetAdjust = Platform.select({ios: 5, default: 20});
+
+        // 9 items to display inline date-time picker, 4 items otherwise
+        const iosNumberOfItems = customTimeSelected ? 9 : 4;
+        const andriodNumberOfItems = customTimeSelected ? 4 : 3;
+        const numberOfItems = Platform.select({ios: iosNumberOfItems, default: andriodNumberOfItems});
+        const COMPONENT_HEIGHT = TITLE_HEIGHT + (numberOfItems * ITEM_HEIGHT) + FOOTER_HEIGHT + bottomSheetAdjust;
+        return [1, COMPONENT_HEIGHT];
+    }, [customTimeSelected]);
+
+    const onSelectTime = useCallback((selectedValue: string) => {
+        setSelectedTime(selectedValue);
+    }, []);
+
+    const handleOnSchedule = usePreventDoubleTap(useCallback(async () => {
+        if (!selectedTime) {
+            logDebug('ScheduledPostOptions', 'No time selected');
+            return;
+        }
+
+        setIsScheduling(true);
+        const schedulingInfo: SchedulingInfo = {
+            scheduled_at: parseInt(selectedTime, 10),
+        };
+
+        const response = await onSchedule(schedulingInfo);
+        setIsScheduling(false);
+
+        if (response?.error) {
+            const errorMessage = response.error as string;
+            showScheduledPostCreationErrorSnackbar(errorMessage);
+        } else {
+            await dismissBottomSheet();
+        }
+    }, [onSchedule, selectedTime]));
+
+    const renderContent = () => {
+        return (
+            <View style={style.container}>
+                {
+                    !isTablet &&
+                    <View style={style.titleContainer}>
+                        <FormattedText
+                            id='scheduled_post.picker.title'
+                            defaultMessage='Schedule draft'
+                            style={style.title}
+                        />
+                    </View>
+                }
+                <View style={style.optionsContainer}>
+                    <ScheduledPostCoreOptions
+                        userTimezone={userTimezone}
+                        onSelectOption={onSelectTime}
+                        onCustomTimeSelected={setCustomTimeSelected}
+                    />
+                </View>
+            </View>
+        );
+    };
+
+    const renderFooter = (props: BottomSheetFooterProps) => (
+        <ScheduledPostFooter
+            {...props}
+            onSchedule={handleOnSchedule}
+            isScheduling={isScheduling}
+        />
+    );
+
+    return (
+        <BottomSheet
+            renderContent={renderContent}
+            componentId={Screens.SCHEDULED_POST_OPTIONS}
+            closeButtonId={SCHEDULED_POST_OPTIONS_BUTTON}
+            snapPoints={snapPoints}
```

Comment:
For this one, it seems that there are times where no options show on screen, this will leave an empty space in the bottom sheet initial snap point, I think you should consider using the dynamic sizing provided by the library for this use case.

Reply:
This can be taken up later, we already have a ticket about bottom sheet and fixing up the height. I will put the link of the conversion there to track.

Reply:
https://mattermost.atlassian.net/browse/MM-63710